tensorflow/core/kernels/reduction_gpu_kernels.cu.h

```diff
@@ -1135,6 +1137,41 @@ struct ReduceFunctor<GPUDevice, Eigen::internal::SumReducer<T>> {
   }
 };
 
+// Specialization for bfloat16 with fp32 accumulation.
+template <>
+struct ReduceFunctor<GPUDevice, Eigen::internal::SumReducer<Eigen::bfloat16>> {
```

Comment:
Out of curiosity, do you know why there is no specialization for float16? I would think float16 should also have a specialization that uses float32.

No need to change any code in response to this comment, I am just wondering.

Reply:
Thanks for the review!

I agree there probably should also be a specialization for float16 Sum like there is for float16 Mean. Maybe it just wasn't added because sum isn't used as often?

I did notice for the CPU kernels there is a bf16 specialization for sum and a [comment ](https://github.com/tensorflow/tensorflow/blob/078a9276f46f95fcacb3778b81294530165d0746/tensorflow/core/kernels/reduction_ops.h#L60-L61) saying the specialization should be done for the rest of the ops too.